components/Navbar/NavbarDropdownComponent.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+import React, { useState } from "react";
+import { Prisma } from "@/lib/prisma";
+
+export type NavbarDropdownProps = Omit<
+  Prisma.NavbarDropdownsGetPayload<{}>,
+  "items"
+> & {
+  items: {
+    title: string;
+    groups_with: string;
+    url: string;
+  }[];
+};
+
+type Item = {
+  title: string;
+  groups_with: string;
+  url: string;
+};
+
+export function NavbarDropdown({ title, items }: NavbarDropdownProps) {
+  const [open, setOpen] = useState(false);
+  const changeState = () => {
+    setOpen(!open);
+  };
+
+  const groupedItems: Item[][] = items.reduce((groups, item) => {
+    if (
+      groups.length === 0 ||
+      groups[groups.length - 1][0].groups_with !== item.groups_with
+    ) {
+      groups.push([item]);
+    } else {
+      groups[groups.length - 1].push(item);
+    }
+    return groups;
+  }, [] as Item[][]);
+
+  return (
+    <>
+      {/* MOBILE */}
+      <div className="md:hidden flex flex-col gap-1">
```

Comment:
Consider using a [Click Away Listener](https://www.npmjs.com/package/react-click-away-listener) to close the nav popup when the user clicks the background.